core/src/main/java/org/elasticsearch/index/mapper/MappedFieldType.java

```diff
@@ -315,6 +315,25 @@ protected BytesRef indexedValueForSearch(Object value) {
         return BytesRefs.toBytesRef(value);
     }
 
+    /** Returns true if the field is searchable.
+     *
+     */
+    public boolean isSearchable() {
+        return indexOptions() != IndexOptions.NONE;
+    }
+
+    /** Returns true if the field is aggregatable.
+     *
+     */
+    public boolean isAggregatable() {
+        try {
+            fielddataBuilder();
+            return true;
+        } catch (Exception e) {
```

Comment:
I've changed the exception in fielddataBuilder and failIfNoDocValues. Depending on the type, they were using an IllegalStateException or an IllegalArgumentException. I've changed this behavior and now they always use an IllegalArgumentException which is mapped on an http error 400 when they are thrown during a request.